core/src/main/scala/io/finch/request/package.scala

```diff
@@ -34,6 +34,55 @@ import scala.reflect.ClassTag
 package object request {
 
   /**
+   * A reusable validation rule that can be applied to any ''RequestReader'' with a matching type.
+   */
+  class ValidationRule[T] private[request] (val description: String, predicate: T => Boolean) extends (T => Boolean) { self =>
```

Comment:
Yes, that should be fine. Do you want me to change that? I'd have to wait 12 hrs before I can get back to work on that. Alternatively, if you have time, you could merge and change it in a follow-up commit, so that I can prepare PR 2 in 12 hrs instead.